packages/core/src/MapIt/index.js

```diff
@@ -157,13 +169,43 @@ function MapIt({
                 return { ...accum, [k]: v };
               }, {});
           }
-          const areaKeys = Object.keys(areaData).join();
-
-          return fetchGeoJson(areaKeys, Object.values(areaData));
+          return areaData;
         });
       });
     },
-    [url, filterCountriesMemoized, drawProfile, geoLevel, fetchGeoJson]
+    [drawProfile, filterCountriesMemoized, geoLevel, url]
+  );
+
+  const loadGeometryForChildLevel = useCallback(
+    async (areaId) => {
+      const childrenData = await fetchChildren(areaId);
+      return fetchGeoJson(
+        Object.keys(childrenData).join(),
+        Object.values(childrenData)
+      );
+    },
+    [fetchChildren, fetchGeoJson]
+  );
+
+  const loadGeometryForGrandChildLevel = useCallback(
+    async (areaId) => {
+      const childrenData = await fetchChildren(areaId);
```

Comment:
@kilemensi  `loadGeometryForChildLevel` isn't fitting here,because after it does fetchChildren which returns only areas details, it goes ahead and fetch geoJson (pulls in features for each key, and merge each properties & area details). This all is not needed on `loadGrandChildren`

Reply:
I meant `childrenData` @KhadijaMahanga not the whole ` loadGeometryForChildLevel` i.e. if both ` loadGeometryForChildLevel` & `loadGeometryForGrandChildLevel` use  `const childrenData = await fetchChildren(areaId);` internally, then this can be exposed at component level so that  `childrenData` is available to both rather than each having to do the fetch.